File: demo/src/main/java/com/lcm/app/data/db/DBManager.java
package com.lcm.app.data.db;


import com.lcm.app.data.entity.db.Dog;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.realm.Realm;
import io.realm.RealmResults;

/**
 * ****************************************************************
 * Author: LCM
 * Date: 2018/4/26 19:43
 * Desc:
 * *****************************************************************
 */
@Singleton
public class DBManager {

    private Realm realm = null;

    @Inject
    public DBManager() {
    }


    public void addDog(Dog dog){
        if(realm == null) realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        realm.copyToRealm(dog);
        realm.commitTransaction();
        close();
    }


    public List<Dog> findAllDog(){
        if(realm == null) realm = Realm.getDefaultInstance();
        RealmResults<Dog> all = realm.where(Dog.class).findAll();
        List<Dog> res = realm.copyFromRealm(all);
        close();
        return res;
    }


    private void close(){
        if(realm != null){
            realm.close();
            realm = null;
        }
    }

}

File: demo/src/main/java/com/lcm/app/data/db/DBManager.kt
package com.lcm.app.data.db



import com.lcm.app.data.entity.db.Dog
import io.realm.Realm
import io.realm.RealmResults
import javax.inject.Inject
import javax.inject.Singleton

/**
 * ****************************************************************
 * Author: LCM
 * Date: 2018/4/26 19:43
 * Desc:
 * *****************************************************************
 */
@Singleton
class DBManager @Inject constructor() {

    private var realm: Realm?=null

    fun addDog(dog:Dog){
        if(realm == null) realm = Realm.getDefaultInstance()
        realm!!.beginTransaction()
        realm!!.copyToRealm(dog)
        realm!!.commitTransaction()
        close()
    }

    fun findAllDog():List<Dog>{
        if(realm == null) realm = Realm.getDefaultInstance()
        val findAll:RealmResults<Dog> = realm!!.where(Dog::class.java).findAll()
        val res =  realm!!.copyFromRealm(findAll)
        close()
        return res
    }

    private fun close(){
        if(realm != null){
            realm!!.close()
            realm = null
        }
    }
}
